### ISSUE TITLE
Inconsistent progress states retrieved from backend

### ISSUE THREAD

**Issue by satsie** [saving-satoshi/saving-satoshi#1061]
I'm not sure what's going on here, but I have two logins with nearly full play throughs. Sometimes when I log in on production, and now even locally, I am locked out of all challenges. 

I'm attaching a [file](https://github.com/user-attachments/files/16752826/progress_response2.txt) of a sample response for `https://api.savingsatoshi.com/progress`. It says my progress is ` "progress": "CH8OUT1",`, but none of the lessons are complete. Example:

```
{
    "account": 609,
    "progress": "CH8OUT1",
    "progress_state": {
        "chapters": [
            {
                "id": 1,
                "lessons": [
                    {
                        "id": "CH1INT1",
                        "path": "/chapter-1/intro-1",
                        "completed": false
                    },
                    {
                        "id": "CH1INT2",
                        "path": "/chapter-1/intro-2",
                        "completed": false
                    },

...

            {
                "id": 2,
                "lessons": [
                    {
                        "id": "CH2INT1",
                        "path": "/chapter-2/intro-1",
                        "completed": false
                    },
                    {
                        "id": "CH2INT2",
                        "path": "/chapter-2/intro-2",
                        "completed": false
                    },

...

                    {
                        "id": "CH9PRP4",
                        "path": "/chapter-9/proposal-4",
                        "completed": false
                    },
                    {
                        "id": "CH9OUT1",
                        "path": "/chapter-9/outro-1",
                        "completed": false
                    }
                ],
                "completed": false,
                "hasDifficulty": false
            }
        ],
        "currentLesson": "CH1INT1",
        "currentChapter": 1
    }
}
```

For those with access to the shared drive, you can find the credentials for the accounts I am using here: https://docs.google.com/document/d/1PPT1OrRRZogcn-FXSk8cE-RQKV5hKkTq8fANRiy9WnQ/edit

One of the accounts was created about a week and a half ago.

Worth mentioning that it seems to be inconsistent. I tried raising this issue in Discord, and after refreshing the page I wasn't locked out of the challenges anymore. Also, I've been doing a lot of work in chapter 6 today and pressing the reset terminal button a lot. 

Locally, if I add `?dev=true` my correct progress history is restored, but I wonder if this has more to do with refreshing.
EDIT: it reverted back to locking me out of everything

I apologize that I have no other steps to reproduce other than just trying to log into prod with the credentials.

**Comment by satsie** [comment]
Did some digging and I think I might be experiencing the same issue as this comment: https://github.com/saving-satoshi/saving-satoshi/pull/968#pullrequestreview-2176700008

After briefly looking at `state/progressState.ts`, is it possible the front end made a PUT call to the backend to set my account(s) to the `defaultProgressState`? @daviroo 

Sorry to make a big stink about this but I am very worried that this could be happening to all users right now. Hopefully it's just me though.

**Comment by benalleng** [comment]
Closed by #1063
